File: 206-reverse-linked-list/reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        prv=None
        curr=head
        next=None

        while curr!= None:
            next=curr.next
            curr.next=prv
            prv=curr
            curr=next
        return prv
        

File: 206-reverse-linked-list/reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        // if(!head || !head->next)return head;
        // ListNode* ans= reverseList(head->next);
        // head->next->next=head;
        // head->next=NULL;
        // return ans;

        ListNode* prv=NULL;
        ListNode* curr=head;
        ListNode* next=NULL;

        while(curr!=NULL){
            next=curr->next;
            curr->next=prv;
            prv=curr;
            curr=next;
        }
        return prv;

    }
};